src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -186,7 +186,10 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
 
             logger.Info(ErrorCode.SiloGcSetting, "Silo starting with GC settings: ServerGC={0} GCLatencyMode={1}", GCSettings.IsServerGC, Enum.GetName(typeof(GCLatencyMode), GCSettings.LatencyMode));
             if (!GCSettings.IsServerGC || !GCSettings.LatencyMode.Equals(GCLatencyMode.Batch))
```

Comment:
The fact that ServerGC does not work on single-core machines seems to be tripping people all the time.
People trying to follow best practices (e.g. enabling ServerGC) should not be under the false impression that they have successfully configured the system - I think we should keep the warning when we are running on single-core systems. 
I have considered whether we should further tune the warning according to the number of cores, but I thought the proposed solution to be sufficient.